pallets/rewards/src/lib.rs

```diff
@@ -220,6 +215,29 @@ pub mod pallet {
 								.saturated_into(),
 							factor: (*factor).saturated_into(),
 						};
+
+						//transfer funds from pallet account to users account
+						ensure!(
+						Self::transfer_pdex_rewards(
+							&Self::get_pallet_account(),
+							&user,
+							reward_info.total_reward_amount
+						)
+						.is_ok(),
+						Error::<T>::TransferFailed
```

Comment:
`transfer_pdex_rewards()` already returns `DispathResult` with proper reason. What's the point of overwriting it here with generic `TransferFailed` error?

Reply:
Removed ensured and passed back the error to the calling function.